gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/BaseEntity.java

```diff
@@ -54,7 +55,7 @@
   private final transient ImmutableSortedMap<String, Value<?>> properties;
   private final K key;
 
-  abstract static class Builder<K extends IncompleteKey, B extends Builder<K, B>> {
+  public abstract static class Builder<K extends IncompleteKey, B extends Builder<K, B>> {
```

Comment:
Yes, because it is exported by public methods, leading to [weirdness](http://googlecloudplatform.github.io/gcloud-java/apidocs/com/google/gcloud/datastore/Entity.html#emptyBuilder()) in the current javadoc.